Add Kotlin to all polyglot pages

mlvandijk opened the pull request:
Added Kotlin to all polyglot pages
- [x] api.md
- [x] configuration.md 
- [x] cucumber-expressions.md
- [x] reporting.md
- [x] step-definitions.md
- [x] anti-patterns.md
- [x] browser-automation.md

mlvandijk committed:
[WIP] Add Kotlin to polyglot

mlvandijk committed:
Merge remote-tracking branch 'origin/master' into add-kotlin

mlvandijk committed:
Add part of tutorial for Kotlin

mlvandijk committed:
Add rest of tutorial for Kotlin

mlvandijk committed:
Add rest of tutorial for Kotlin

mlvandijk committed:
Update main page

mlvandijk committed:
WIP Add kotlin to polyglot pages

mlvandijk committed:
Merge remote-tracking branch 'origin/master' into more-kotlin

# Conflicts:
#	content/cucumber/api.md
#	content/cucumber/reporting.md
#	content/guides/10-minute-tutorial.md

mlvandijk committed:
WIP Add kotlin to polyglot pages

mlvandijk committed:
Update contributing to docs for writing style, shortcodes and Kotlin

mlvandijk committed:
Update contributing to docs for writing style, shortcodes and Kotlin

(cherry picked from commit 74fdf82)

mlvandijk committed:
Add Kotlin to page

(cherry picked from commit 83e1310)

mlvandijk committed:
Add Kotlin to page

Add Kotlin to page and update java examples to cucumber expressions
Add Kotlin to pages

mlvandijk committed:
Add Kotlin to page and update java examples to cucumber expressions

(cherry picked from commit fcb5c80)

mlvandijk committed:
Add Kotlin to pages

(cherry picked from commit f11e416)

mlvandijk committed:
Add kotlin to page (partial)

(cherry picked from commit d310d98)

mlvandijk committed:
Merge branch 'master' into more-kotlin

# Conflicts:
#	content/cucumber/api.md
#	content/cucumber/configuration.md
#	content/cucumber/step-definitions.md

luke-hill commented on content/community/contributing-to-documentation.md:
```diff
@@ -123,20 +94,24 @@ A language select will be displayed if the page specifies the following in the f
 * Ruby
+* Kotlin (optional)
 * .Net (optional; some pages only)
+* Kotlin (optional)
```
Is there a reason this is mentioned twice above and here?

mlvandijk replied:
One is an example of the polyglot front matter, and one is a list of the currently supported languages. Is that what you mean?

mpkorstanje replied:
I think he means, why does kotlin occur four times in two lists? Shouldn't it occur only twice in two lists?

mlvandijk replied:
Yes, found this too. Fixed it ;)

mlvandijk committed:
Fix duplication

mlvandijk committed:
Remove sentence as it makes no sense here

mlvandijk committed:
Add todo and fix formatting

mlvandijk committed:
Merge branch 'cucumber-master' into more-kotlin

# Conflicts:
#	content/cucumber/configuration.md

mlvandijk committed:
Add Kotlin example for browser automation

mlvandijk committed:
Revert "Add Kotlin example for browser automation"

This reverts commit 330a0e2

mlvandijk committed:
Add Kotlin examples throughout page

mlvandijk committed:
Add TODO for Kotlin example

mlvandijk committed:
Add TODO for Kotlin example

mlvandijk committed:
Add Kotlin example for browser automation

mlvandijk committed:
Add Kotlin example for type registry

mlvandijk committed:
Merge remote-tracking branch 'origin/master' into more-kotlin

# Conflicts:
#	content/community/contributing.md

mlvandijk committed:
Updates based on Kotlin review comments

mlvandijk committed:
Clarify IntelliJ translation to Kotlin code

mlvandijk committed:
Fix syntax for Hooks

olleolleolle commented:
This is beginning to look _complete_. That's cool!

mlvandijk commented:
@olleolleolle Afaik I added Kotlin to all polyglot pages and added a Kotlin translation for all code examples. Please let me know if you think there are any missing?
Also, a friend reviewed the Kotlin code and I did some fixes based on his review ;)
So imho this PR is good to go!

olleolleolle commented:
I think we are ready for a merge.

olleolleolle reviewed (approved):
LGTM

mlvandijk commented:
Thanks @olleolleolle

mpkorstanje commented on content/guides/browser-automation.md:
```diff
@@ -77,6 +77,48 @@ public class ExampleSteps {
+
+class ExampleSteps: En {
+
+    lateinit var driver: WebDriver
```
Why `lateinit`?

mlvandijk replied:
`lateinit `prevents having to initialize here, and making WebDriver nullable.
Personally, I prefer `lateinit var driver: WebDriver` to `private var driver: WebDriver? = null`

mpkorstanje replied:
But shouldn't the webdriver be injected then? It is never instantiated.
